In [1]:
# import necessary libraries
import pandas as pd
import numpy as np

Importing data

In [3]:
df = pd.read_csv('2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')

Now, let's have a glimpse of the data.

In [5]:
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB


Next, let's check for the level of completeness in the data.

In [6]:
df.isnull().sum()

order_id          0
shop_id           0
user_id           0
order_amount      0
total_items       0
payment_method    0
created_at        0
dtype: int64

Great, we do not have any missing data! Now let's access what is wrong with the calculation.

In [7]:
df['order_amount'].describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

#### Observation;
The mean order amount is 3145.13 which is way too high for a sneaker storeThis possibly indicates that  stores data probably has outliers implying one of the case scenarios there was human error in keyeing data or fraudlent cases.

#### Metric to be used:
The median on the other hand is 284 which is much more reasonable and  therefore a better metric to be used to access AOV.

#### What is its value?

In [9]:

df['order_amount'].median()

284.0

A better value to report for AOV on this dataset is the median of order amount which is $284.00.